File: 713-subarray-product-less-than-k/713-subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int product = 1;
        int count = 0;
        int left = 0;
        for(int right = 0; right < nums.length; right++) {
            product *= nums[right];
            while(left <= right && product >= k) {
                product /= nums[left];
                left += 1;
            }
            count += right - left + 1;
        }
        return count;
    }
}

File: 713-subarray-product-less-than-k/713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        prod = 1
        l = 0
        c = 0
        for r in range(len(nums)):
            prod *= nums[r]
            while l <= r and prod >= k:
                prod //= nums[l]
                l += 1
            c += r - l + 1
        return c